src/EnergyPlus/OutputReportTabular.cc

```diff
@@ -11198,6 +11250,276 @@ namespace OutputReportTabular {
+            for (int j = 0; j < columnNum - 1; j++) {
+                double curValue = ZoneBins(i)[j];
+                if (curValue > columnMax[j]) columnMax[j] = curValue;
+                if (curValue < columnMin[j]) columnMin[j] = curValue;
+                columnSum[j] += curValue;
+                tableBody(j + 1, i) = RealToStr(curValue, 2);
+            }
+            std::string startDateTime = DateToString(int(ZoneBins(i)[columnNum - 1]));
+            tableBody(columnNum, i) = startDateTime;
+        }
+        rowHead(NumOfZones + 1) = "Min";
+        rowHead(NumOfZones + 2) = "Max";
+        rowHead(NumOfZones + 3) = "Average";
+        for (int j = 0; j < columnNum - 1; j++) {
+            tableBody(j + 1, NumOfZones + 1) = RealToStr(columnMin[j], 2);
+            tableBody(j + 1, NumOfZones + 2) = RealToStr(columnMax[j], 2);
+            tableBody(j + 1, NumOfZones + 3) = RealToStr(columnSum[j] / NumOfZones, 2);
+        }
+        tableBody(columnNum, NumOfZones + 1) = "-";
+        tableBody(columnNum, NumOfZones + 2) = "-";
+        tableBody(columnNum, NumOfZones + 3) = "-";
+
+        WriteTable(tableBody, rowHead, columnHead, columnWidth);
+        if (sqlite) {
+            sqlite->createSQLiteTabularDataRecords(tableBody, rowHead, columnHead, sqlTitle, "Entire Facility", tblSubTitle);
+        }
+        if (ResultsFramework::OutputSchema->timeSeriesAndTabularEnabled()) {
+            ResultsFramework::OutputSchema->TabularReportsCollection.addReportTable(
+                    tableBody, rowHead, columnHead,  tblTitle, "Entire Facility", tblSubTitle);
+        }
+    }
+
+    void WriteThermalResilienceTables()
+    {
+
+        // Using/Aliasing
+        using DataHeatBalFanSys::ZoneHeatIndexHourBins;
+        using DataHeatBalFanSys::ZoneHeatIndexOccuHourBins;
+        using DataHeatBalFanSys::ZoneHumidexHourBins;
+        using DataHeatBalFanSys::ZoneHumidexOccuHourBins;
+        using DataHeatBalFanSys::ZoneLowSETHours;
+        using DataHeatBalFanSys::ZoneHighSETHours;
+
+        if (NumOfZones > 0) {
+            std::string tblTitle = "Annual Thermal Resilience Summary";
+            std::string sqlTitle = "AnnualThermalResilienceSummary";
+            WriteReportHeaders(tblTitle, "Entire Facility", OutputProcessor::StoreType::Averaged);
+
+            int columnNum = 5;
+            std::string subTitle = "Heat Index Hours";
+            std::vector<std::string> columnHeadNames = { "Safe (≤ 80°F) [Hours]",
+                                                         "Caution (80, 90°F] [Hours]",
+                                                         "Extreme Caution (90, 103°F] [Hours]",
+                                                         "Danger (103, 125°F] [Hours]",
+                                                         "Extreme Danger (> 125°F) [Hours]" };
+            WriteResilienceBinsTable(columnNum, tblTitle, subTitle, sqlTitle, columnHeadNames, ZoneHeatIndexHourBins);
```

Comment:
@xuanluo113 This seems like the hard way to build a new table report instead of using the `OutputReportPredined` functions like `newPreDefReport` and `newPreDefColumn`.
Is this hard-wired to always write deg F and other data in deg C below? Output tables should be sensitive to units conversion options specified in OutputControl:Table:Style. @JasonGlazer ?

Reply:
I was following the style of [WriteAdaptiveComfortTable](https://github.com/NREL/EnergyPlus/blob/1426e755443ea4f8fe19497d4c7be554ac27a048/src/EnergyPlus/OutputReportTabular.cc#L11157) first and was not aware of the `OutputReportPredined` style. I could clean them up to follow the predefined style then.

There's nothing to do with deg F or C. The standard [equations](https://www.wpc.ncep.noaa.gov/html/heatindex_equation.shtml) and [thresholds](https://www.weather.gov/ama/heatindex) are in deg F, so the table headers are written this way. The calculation results are converted back to [deg C](https://github.com/NREL/EnergyPlus/blob/1426e755443ea4f8fe19497d4c7be554ac27a048/src/EnergyPlus/HeatBalanceSurfaceManager.cc#L4963) at the time step loop.

Reply:
The column headings having ranges of values in F is an unusual case. I would think the SI version of the table the values should be shown in ranges of C instead of F. You might need to be sensitive to the unit system requested by the user and change the values when defining the table.

Reply:
@JasonGlazer Thanks for the clarification. I'll convert the headers to deg C thresholds. 
Regarding the predefined table style, if I want to add Zone Min, Max, Average and Sum of a runtime variable to the table end, is it the best practice to calculate these aggregation values in the [FillRemainingPredefinedEntries](https://github.com/NREL/EnergyPlus/blob/1426e755443ea4f8fe19497d4c7be554ac27a048/src/EnergyPlus/OutputReportTabular.cc#L6447) function?